pkg/operator/ceph/object/user/controller.go

```diff
@@ -549,16 +642,25 @@ func (r *ReconcileObjectStoreUser) reconcileCephUserSecret(cephObjectStoreUser *
 	// Generate Kubernetes Secret
 	secret := r.generateCephUserSecret(cephObjectStoreUser, userConfig, tlsSecretName)
 
-	// Set owner ref to the object store user object
-	err := controllerutil.SetControllerReference(cephObjectStoreUser, secret, r.scheme)
-	if err != nil {
-		return reconcile.Result{}, errors.Wrapf(err, "failed to set owner reference of ceph object user secret %q", secret.Name)
-	}
+	// only manage the secret if there are no explicit keys
+	if len(cephObjectStoreUser.Spec.Keys) == 0 {
```

Comment:
I don't see an issue with continuing to create the existing reference secret here. The overall system will be simpler if we don't have conditionals on when we do or don't create this. If users don't want to use it, they don't have to.

Reply:
That would still be stable output until the config changes, and I think that's fine. The changes here are big enough without this thrown in the mix, and I'd prefer to simplify what we can.

Reply:
I added a `.spec.disableAutomaticSecret` field which, obviously, defaults to to `false` and will retain the automatically generated secret by default.  In my env, I would prefer it isn't present to avoid confusion with the source of truth explicit secrets. This help keep the listing of secrets down for manual inspection when there are 10s or 100s of CRs.